### ISSUE TITLE
Conversion of -0.0 to string doesn't include sign as PHP does

### ISSUE THREAD

**Issue by jesseschalken** [facebook/hhvm#7425]
### HHVM Version

```
HipHop VM 3.16.0-dev (rel)
Compiler: heads/master-0-ge9eb3fc414fbae9b2c999a4f11390e23050b23b7
Repo schema: 9cc0ebe15bc9d2ba3609c5e4458deeef9ed86dcc
```
### Standalone code, or other way to reproduce the problem

``` php
<?php

print 0.0 * -1.0;
```

(Note that `-0.0` doesn't produce negative zero PHP < 7.0.2, see https://bugs.php.net/bug.php?id=52355 and https://3v4l.org/Taoar.)
### Expected result

```
-0
```
### Actual result

```
0
```

See https://3v4l.org/HmkTX

**Comment by jesseschalken** [comment]
Note this is both a PHP5 and PHP7 incompatibility (#7426 is just a PHP7 incompatibility, though).

**Comment by jesseschalken** [comment]
Looks like this was fixed in 3.15.4: https://3v4l.org/HmkTX
